src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -141,39 +121,75 @@ function _buildybus(branches, nodes, fixed_admittances)
         _ybus!(ybus, b, num_bus)
 
     end
-
     for fa in fixed_admittances
         _ybus!(ybus, fa, num_bus)
     end
+    return SparseArrays.dropzeros!(ybus)
+end
 
-    return ybus
-
+function _goderya(ybus::SparseArrays.SparseMatrixCSC{ComplexF64, Int64})
+    node_count = size(ybus)[1]
+    max_I = node_count^2
+    I, J, val = SparseArrays.findnz(ybus)
+    T = SparseArrays.sparse(I, J, ones(length(val)))
+    T_ = T * T
+    for n in 1:(node_count - 1)
+        I, _, _ = SparseArrays.findnz(T_)
+        if length(I) == max_I
+            @info "The System has no islands"
```

Comment:
This just checks for disconnected leaf nodes, right? If that's the case, that's a good check, but the error message should be "The System has no disconnected buses". If I'm wrong, you can disregard.

Reply:
Goderya's algorithm searches for islands generally. See the tests.